IPython/terminal/shortcuts.py

```diff
@@ -123,6 +124,18 @@ def _following_text():
         _following_text_cache[pattern] = condition
         return condition
 
+    @Condition
+    def not_inside_unclosed_string():
+        app = get_app()
+        ver = f"{sys.version_info.major}.{sys.version_info.minor}"
+        g = parso.load_grammar(version=ver)
+        parser = g.parse(app.current_buffer.document.current_line_before_cursor)
```

Comment:
Wow, that was fast, I barely had time to look into the parso docs !

```suggestion
        g = parso.load_grammar()
        parser = g.parse(app.current_buffer.document.current_line_before_cursor)
```

I think the default is to used current version number.

Reply:
I hadn’t used it before, but the docs are really good! And you’re right, didn’t need to do that.

Reply:
also, `load_grammar` isn't free: probably worth caching this reference, as it can likely be used for other things.